ctypesgen/printer_python/printer.py

```diff
@@ -245,36 +273,14 @@ def print_struct_members(self, struct):
                 aligned = aligned.evaluate(None)
             self.file.write("{}_{}._pack_ = {}\n".format(struct.variety, struct.tag, aligned))
 
-        # handle unnamed fields.
-        unnamed_fields = []
-        names = set([x[0] for x in struct.members])
-        anon_prefix = "unnamed_"
-        n = 1
-        for mi in range(len(struct.members)):
-            mem = list(struct.members[mi])
-            if mem[0] is None:
-                while True:
-                    name = "%s%i" % (anon_prefix, n)
-                    n += 1
-                    if name not in names:
-                        break
-                mem[0] = name
-                names.add(name)
-                if type(mem[1]) is CtypesStruct:
-                    unnamed_fields.append(name)
-                struct.members[mi] = mem
-
-        self.file.write("%s_%s.__slots__ = [\n" % (struct.variety, struct.tag))
-        for name, ctype in struct.members:
-            self.file.write("    '%s',\n" % name)
-        self.file.write("]\n")
-
+        unnamed_fields = struct.unnamed_fields
         if len(unnamed_fields) > 0:
             self.file.write("%s_%s._anonymous_ = [\n" % (struct.variety, struct.tag))
             for name in unnamed_fields:
                 self.file.write("    '%s',\n" % name)
             self.file.write("]\n")
 
+        # TODO part of this could be deduplicated
```

Comment:
indeed, fields and annotations effectively contain the same info, so one should be derived from the other.

Reply:
Yeah, I'm thinking about adding a dict to each class that contains the bit field information and setting it afterwards.

Reply:
Update: Done. See https://github.com/zwergziege/ctypesgen/tree/reduce_struct_field_duplication

Reply:
The branch is still using the decorator, which we figured is unnecessary. Can you update that?
Also it's not clear to me what the branch is actually trying to achieve, or why it should require python 3.10?
Looks a bit unfinished?

Reply:
> The branch is still using the decorator, which we figured is unnecessary. Can you update that?

Done

> [...] the requirement would affect the generated bindings file, which is a problem and not mergeable in the foreseeable future IMO.

Ah I forgot that this is in the generated file... That's problematic. I changed it such that it uses `typing.get_type_hints` instead, which is already supported in 3.7, so I guess that should be fine.